megamek/src/megamek/client/ui/swing/UnitSelectorDialog.java

```diff
@@ -706,17 +703,16 @@ void refreshUnitView() {
             // error unit didn't load right. this is bad news.
             populateTextFields = false;
         }
-        if (populateTextFields && (mechView != null)) {
+        if (populateTextFields) {
```

Comment:
As indicated by the comment above, if an exception is thrown while loading the unit, mechView will still not be initialized. The exception is already caught and logged, but removing this null check will cause another exception to be thrown which may cause the ui to stop responding.

Reply:
If you look under that comment, it sets populateTextFields to false. That will also trigger this IF statement as false, so the mechview != null check is not required.

Reply:
Sure enough